src/services/text.ts

```diff
@@ -19,3 +21,16 @@ export const copyText = async (text: string, buttonRef: React.MutableRefObject<H
     textarea.remove();
   }
 };
+
+export function capitalizeFirstLetter(input: string) {
+  if (input.length) {
+    return input?.charAt(0).toUpperCase() + input.slice(1);
+  }
+
+  logger.warn('invalid string argument');
+  return input;
+}
+
+export function truncateText(input: string, length: number, ellipsis: boolean) {
+  return input.substring(0, length) + (ellipsis && input.length > length ? '...' : '');
```

Comment:
```suggestion
  return input.substring(0, length) + (ellipsis && input.length > length ? '&mldr;' : '');
```

or

```suggestion
  return input.substring(0, length) + (ellipsis && input.length > length ? '…' : '');
```

https://www.compart.com/en/unicode/U+2026